src/sky/pro/Main.java

```diff
@@ -0,0 +1,129 @@
+package sky.pro;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+
+public class Main {
+
+
+    public static void main(String[] args) {
+
+        // write your code here
+        //task 1
+        printYear(2019);
+        //task 2
+        printVersionApp(2020, 1);
+        // task 3
+        System.out.println("Потребуется дней: " + getTimeDelivery(90));
+        //task 4
+        searchDuplicates("aabccddefgghiijjkk");
+        //task 5
+        reverseArray(new int[]{3, 2, 1, 6, 5});
+        //task 6
+        int [] arr=generateRandomArray();
+        double sum= calculateTotalSum(generateRandomArray());
+        double averageAmount= calculateAverageSum(sum,arr);
+        printAverageSum(averageAmount);
+    }
+
+    public static void printYear(int year) {
+        // write your code here
+        if (year % 4 == 0 && year % 100 != 0 || year % 400 == 0) {
+            System.out.println(year + " год является високосным");
+        } else {
+            System.out.println(year + " год не является високосным");
+        }
+    }
+
+    public static void printVersionApp(int year, int clientOC) {
+        // write your code here
+        int currentYear = LocalDate.now().getYear();
+        boolean isNeededLite = year >= currentYear;
+        boolean isАndroid = clientOC == 1;
+        if (isАndroid) {
+            if (isNeededLite) {
+                System.out.println("Установите lite-версию приложения для Аndroid по ссылке.");
+            } else {
+                System.out.println("Установите версию приложения для Аndroid по ссылке.");
+
+            }
+        } else {
+            if (isNeededLite) {
+                System.out.println("Установите lite-версию приложения для iOS по ссылке.");
+            } else {
+                System.out.println("Установите версию приложения для iOS по ссылке.");
+
+            }
+        }
+    }
+
+    public static int getTimeDelivery(int deliveryDistance) {
+        int timeDelivery = 1;
+        if (deliveryDistance < 20) {
+            timeDelivery = timeDelivery;
+        } else if (deliveryDistance >= 20 && deliveryDistance < 60) {
```

Comment:
поскольку условный оператор единый через else if, то обратные условия от предыдущих можно не писать, поскольку они автоматом выполняются, если мы попали на эту ветку
```suggestion
        } else if (deliveryDistance < 60) {
```